src/machine/asm/execute.S

```diff
@@ -96,6 +96,39 @@
 #define MOV_RS2r_TO_RDX movq RS2r, %rdx
 #define MOV_RDX_TO_RS2r movq %rdx, RS2r
 
+#define ZEROED_MEMORY(frame_reg) \
+  movq frame_reg, %rdi; \
+  shl $CKB_VM_ASM_MEMORY_FRAME_SHIFTS, %rdi; \
+  addq $CKB_VM_ASM_ASM_CORE_MACHINE_OFFSET_MEMORY, %rdi; \
+  addq MACHINE, %rdi; \
+  xor %eax, %eax; \
+  movl $CKB_VM_ASM_MEMORY_FRAMESIZE, %ecx; \
+  shr $2, %ecx; \
+  cld; rep; stosl;
+
+#define INITED_MEMORY(address_reg, temp_reg1, temp_reg2, temp_reg2d, length) \
```

Comment:
I'm concerned about the duplication of all the code here, can you get some stat number on the code size before and after this change for the whole `execute.o` section? We do want all the code here to fit in L1 cache so as to be as performant as possible.

Reply:
Also in the worse case, the code for `CHECK_WRITE_PERMISSION` are mostly the same, we should be able to combine them for less code size and less executed instructions.

Reply:
> Also in the worse case, the code for `CHECK_WRITE_PERMISSION` are mostly the same, we should be able to combine them for less code size and less executed instructions.

Yes, the code for checking `out of bounds` is duplicated. I am going to remove this in `CHECK_WRITE_PERMISSION`.

Reply:
> I'm concerned about the duplication of all the code here, can you get some stat number on the code size before and after this change for the whole `execute.o` section? We do want all the code here to fit in L1 cache so as to be as performant as possible.

I made a comparison, the original `execute.o` is 15K, now (based on the latest commit) it is 18K.

Reply:
Let's merge `CHECK_WRITE_PERMISSION` and `INITED_MEMORY` for less instructions executed. Also keep in mind that `INITED_MEMORY` might also be used by memory reading.

Reply:
I have combined the `check_write()` and `inited_meory` in one function named `access_control()`, please take a look.

commit: 

- https://github.com/nervosnetwork/ckb-vm/pull/107/commits/088176fb8346e98bbe21459f4f78e5607cbb4683
- https://github.com/nervosnetwork/ckb-vm/pull/107/commits/7ce3db3375eea0d7f21363963c038279c9a0a653